crates/diesel_models/src/payment_attempt.rs

```diff
@@ -316,10 +332,29 @@ pub struct PaymentAttemptUpdateInternal {
     unified_message: Option<Option<String>>,
 }
 
+impl PaymentAttemptUpdateInternal {
+    pub fn populate_derived_fields(self, source: &PaymentAttempt) -> Self {
+        let mut update_internal = self;
+        update_internal.net_amount = Some(
+            update_internal.amount.unwrap_or(source.amount)
+                + update_internal
+                    .surcharge_amount
+                    .or(source.surcharge_amount)
+                    .unwrap_or(0)
```

Comment:
Here both, amount and surcharge amount are in the lowest denominations ( cents for USD ) right?

Reply:
Yes